### PR TITLE
Feat/app example

### PR DESCRIPTION
**Pull request by rgommezz**
Basic Example, nothing fancy.

- [x] Create example with other APIs
- [x] Update documentation
- [x] Double check override in kotlin, make it work in all versions: This is not backwards compatible as per my tests. Kotlin yells with versions prior to 0.47 during the build with the below error:

`createJSModules' hides member of supertype 'ReactPackage' and needs 'override' modifier`

### CODE REVIEW

**Comment by rgommezz** [comment]
@satya164 I don't know how to solve the backwards compatibility with Kotlin, maybe you have some ideas. Apparently the solution works properly in Java. Maybe we can write that boilerplate file in Java and leave the one with the logic with Kotlin?

Also, if you wanna improve the example with different colors/images, feel free to take over this one or create a different one once it's merged. That way, I can in the meantime work on the medium article 🤖

**Comment by satya164** [comment]
I've never really written kotlin, but I'll give it a try today.

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+  "parser": "babel-eslint",
```
Comment (satya164) [inline .eslintrc]: We should use `eslint-config-callstack-io` instead

File: .flowconfig [hunk 2]
```diff
@@ -39,6 +39,9 @@
 ; Ignore react-native-mock
 .*/node_modules/react-native-mock/*
 
+; Ignore the example app
```
Comment (satya164) [inline .flowconfig]: We shouldn't ignore the example app. Using Flow in examples (and tests) makes sure Flow is not broken for our public API.

File: example/package.json [hunk 3]
```diff
@@ -22,5 +25,8 @@
   },
   "jest": {
     "preset": "react-native"
+  },
+  "eslintConfig": {
+    "extends": "../.eslintrc"
```
Comment (satya164) [inline example/package.json]: It shouldn't be necessary since we don't have any overrides

**Review by satya164 — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 1 above)
Comment (rgommezz) [reply .eslintrc]: There is a separate issue for that: https://github.com/callstack-io/react-native-material-palette/issues/29

**Pull request merged**